tests/stubs/devices/services/temperature.go

```diff
@@ -0,0 +1,73 @@
+package services
+
+import (
+	"crypto/tls"
+	"fmt"
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	"github.com/paypal/gatt"
+	"log"
+	"time"
+)
+
+type Token interface {
+	Wait() bool
+	WaitTimeout(time.Duration) bool
+	Error() error
+}
+
+var TokenClient Token
+var ClientOpts *MQTT.ClientOptions
+var Client MQTT.Client
+var mqttServer string
+
+func HubClientInit(server, clientID, usename, password string) *MQTT.ClientOptions {
+	opts := MQTT.NewClientOptions().AddBroker(server).SetClientID(clientID).SetCleanSession(true)
+	if usename != "" {
+		opts.SetUsername(usename)
+		if password != "" {
+			opts.SetPassword(password)
+		}
+	}
+	tlsConfig := &tls.Config{InsecureSkipVerify: true, ClientAuth: tls.NoClientCert}
+	opts.SetTLSConfig(tlsConfig)
+	return opts
+}
+
+func MqttConnect() {
+	mqttServer = "tcp://<machineip where broker is running>:1883"
+	ClientOpts = HubClientInit(mqttServer, "mockbluetoothserver", "", "")
+	Client = MQTT.NewClient(ClientOpts)
+	if TokenClient = Client.Connect(); TokenClient.Wait() && TokenClient.Error() != nil {
+		fmt.Println("client.Connect Error %s", TokenClient.Error())
+	}
+}
+
+func NewTemperatureService() *gatt.Service {
+	temp := 36.66
+	s := gatt.NewService(gatt.MustParseUUID("09fc95c0-c111-11e3-9904-0002a5d5c51b"))
+	//Adding a characteristic for reading temperature data
+	s.AddCharacteristic(gatt.MustParseUUID("11fac9e0-c111-11e3-9246-0002a5d5c51b")).HandleReadFunc(
+		func(rsp gatt.ResponseWriter, req *gatt.ReadRequest) {
+			fmt.Fprintf(rsp, "Temperature: %f", temp)
+		})
+
+	// Adding characteristic for writing to device
+	s.AddCharacteristic(gatt.MustParseUUID("16fe0d80-c111-11e3-b8c8-0002a5d5c51b")).HandleWriteFunc(
+		func(r gatt.Request, data []byte) (status byte) {
+			log.Println("Wrote:", string(data))
+			publishTopic := "$ke/device/bluetooth-mapper/sensortagmock/datawrite"
```

Comment:
please remove the publish code from here. we should not do more operations other than storing the written value in some internal in-memory variable.

Reply:
Updated